### PR TITLE
Restrict migration tasks to single host.

### PR DESCRIPTION
**Pull request by smarnach**
Note: this PR was filed on behalf of @xitij2000.

We want to run deployments on multiple servers in parallel with identical Ansible configurations. The migrations can't run concurrently on multiple servers, since this would lead to conflicts, so we want to restrict them to a single server. If the migrations fail, the playbook should stop for all servers, which it does with this patch applied.

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @smarnach! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=4243&repo=edx%2Fconfiguration"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: playbooks/roles/edx_django_service/tasks/main.yml [hunk 1]
```diff
@@ -111,6 +111,7 @@
   become_user: "{{ edx_django_service_user }}"
   environment: "{{ edx_django_service_migration_environment }}"
   when: migrate_db is defined and migrate_db|lower == "yes"
+  run_once: yes
```
Comment (smarnach) [inline playbooks/roles/edx_django_service/tasks/main.yml:114]: The "post-migrate commands" (i.e. the task following this one) should also only run on a single host – basically everything that's tagged "migrate".

**Review by smarnach — COMMENTED** [review]
@xitij2000 There is also a migration task in the edxapp role – it's in the file `playbooks/roles/edxapp/tasks/service_variant_config.yml`.  That task should also be marked `run_once: true`.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @smarnach! I've created [OSPR-2051](https://openedx.atlassian.net/browse/OSPR-2051) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by smarnach** [comment]
:+1: This looks good to me, and I tested that normal deployments to a single host still work with this PR applied.

**Comment by smarnach** [comment]
I can't formally approve this PR since I'm the author.

**Review by UmanShahzad — APPROVED** [review]

**Comment by nedbat** [comment]
@edx/devops Please take a look.

**Comment by feanil** [comment]
This looks pretty reasonable, I've schedule review and testing from our side for next sprint(starting next week and going for two weeks).

**Review by nadeemshahzad — APPROVED** [review]

**Comment by nadeemshahzad** [comment]
Thanks @smarnach for this PR. I have tested the changes on sandbox and it looks good to me. I have approved and merged this PR.
  https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/7958/

**Comment by openedx-webhooks** [comment]
@smarnach 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/4243&entry.1484655318=https://github.com/smarnach&entry.752974735=2017-12-07+18:54&entry.1917517419=2018-01-05+12:01&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by smarnach** [comment]
@nadeemshahzad Thanks for the review and for merging!

**Pull request merged**